pkg/pub_integration/lib/src/test_scenario.dart

```diff
@@ -68,7 +68,7 @@ abstract class TestUser {
 
   /// Get a headless browser where this test user is signed-in to their
   /// Google Account.
-  Browser get browser;
+  HeadlessEnv get browser;
```

Comment:
How about just exposing:
```dart
Future<T> withBrowserPage<T>(FutureOr<T> Function(Page page) fn);
```

Do test scenario authors have any business calling `Browser.close()` anyways?


And perhaps make `serverErrors` and `clientErrors` expando properties on `Page`?

Ideally, then you can call `page.serverErrors.clear()` before returning from `withBrowserPage` if intentionally making a test that has a failure.

Reply:
Done with the rename. As discussed offline, we don't need to expose serverErrors or clientErrors for the test cases.